### PR TITLE
AAP-31270: Option to disable cert verification of WCA endpoint

### PR DESCRIPTION
**Pull request by hasys**
Jira Issue: <https://issues.redhat.com/browse/AAP-31270>

### Steps to test
1. Pull down the PR
2. ...
3. ...

## Production deployment

- [ ] This code change is ready for production on its own
- [ ] This code change requires the following considerations before going to production:

### CODE REVIEW

File: ansible_ai_connect/ai/api/model_client/tests/test_wca_client.py [hunk 1]
```diff
@@ -1183,43 +1187,56 @@ def test_get_model_id_without_setting(self):
...
+        "prompt": f"{prompt}\n",
+    }
+    request_headers = {
+        "Authorization": f"ZenApiKey {token}",
+        WCA_REQUEST_ID_HEADER: suggestion_id,
+    }
+    model_client = WCAOnPremClient(inference_url="https://example.com")
 
-        model_client.infer(
+    def test_headers(self):
```
Comment (manstis) [inline ansible_ai_connect/ai/api/model_client/tests/test_wca_client.py:1211]: Should you not add `@override_settings(ANSIBLE_AI_MODEL_MESH_API_VERIFY_SSL=True)`?

If somebody runs this test locally with the setting `False` the test will fail.

File: ansible_ai_connect/ai/api/model_client/tests/test_wca_client.py [hunk 2]
```diff
@@ -1183,43 +1187,56 @@ def test_get_model_id_without_setting(self):
...
+        self.model_client.session.post.assert_called_once_with(
+            "https://example.com/v1/wca/codegen/ansible",
+            headers=self.request_headers,
+            json=self.codegen_data,
+            timeout=None,
+            verify=True,
         )
 
-        model_client.session.post.assert_called_once_with(
+    @override_settings(ANSIBLE_AI_MODEL_MESH_API_TIMEOUT=None)
```
Comment (manstis) [inline ansible_ai_connect/ai/api/model_client/tests/test_wca_client.py:1226]: Should you not add `@override_settings(ANSIBLE_AI_MODEL_MESH_API_VERIFY_SSL=False)`?

**Review by manstis — COMMENTED** [review]
A couple of minor comments for your consideration.

File: ansible_ai_connect/ai/api/model_client/tests/test_wca_client.py (hunk 2 above)
Comment (hasys) [reply ansible_ai_connect/ai/api/model_client/tests/test_wca_client.py]: Fixed, thank you!

**Comment by hasys** [comment]
Hi @manstis , Great comments! Thank you, fixed.

**Review by manstis — APPROVED** [review]
LGTM :+1: Thank-you.

**Pull request merged**